application/src/main/java/org/togetherjava/tjbot/commands/moderation/WarnCommand.java

```diff
@@ -0,0 +1,193 @@
+        // To prevent people from get users warn, only users with
+        // elevated permissions are allowed to use this command
+        if (handleHasPermissions(author, event, event.getGuild()).equals(false)) {
+            return;
+        }
+
+        User target = userOption.getAsUser();
+        try {
+            Optional<String> value = database.read(context -> {
+                try (var select = context.selectFrom(Warns.WARNS)) {
+                    return Optional
+                        .ofNullable(select.where(Warns.WARNS.USERID.eq(target.getId())).fetchOne())
+                        .map(WarnsRecord::getTimeswarned);
+                }
+            });
+
+            if (value.isEmpty()) {
+                event
+                    .replyEmbeds(new EmbedBuilder().setTitle("A behaved user")
+                        .setDescription(
+                                "No warns can be found for the user " + target.getAsTag() + ".")
+                        .setColor(Color.magenta)
+                        .build())
+                    .queue();
+                return;
+            }
+
+            event
+                .replyEmbeds(new EmbedBuilder()
+                    .setAuthor("This command was created by " + author.getEffectiveName(), null,
+                            author.getEffectiveAvatarUrl())
+                    .setTitle("The User " + target.getAsTag() + " warns")
+                    .setDescription("They have " + value.orElseThrow() + " number of warns")
+                    .setColor(Color.magenta)
+                    .build())
+                .queue();
+        } catch (DatabaseException e) {
+            logger.error("Failed to get the number of warns ", e);
+            event.reply("Sorry, something went wrong.").setEphemeral(true).queue();
+        }
+    }
+
+    /**
+     * Handles {@code /warn warn_user user reason} command. Saves the value under the given user,
+     * given guild and add +1 to the number of warns the user has.
+     * <p>
+     * This command can only be used by users with the {@code BAN_MEMBERS} permission.
+     *
+     * @param event the event of the command
+     */
+    private void handleWarnCommand(@NotNull CommandInteraction event) {
+        OptionMapping userOption =
+                Objects.requireNonNull(event.getOption(WARN_USER_OPTION), "The target is null");
+        Member author = userOption.getAsMember();
+        Guild guild = Objects.requireNonNull(event.getGuild());
+
+        // To prevent people from using warn content, only users with
+        // elevated permissions are allowed to use this command
+        if (handleHasPermissions(author, event, guild).equals(false)) {
+            return;
+        }
+
+        // /warn warn_user @Zabuzard who knows
+        User target = userOption.getAsUser();
+        String reason = Objects.requireNonNull(event.getOption(WARN_REASON_OPTION)).getAsString();
+        // TODO double check this part
+        int times = +1;
```

Comment:
* bad variable name, "times"-what? `warnings`? `warningAmount`, `warningCount`?
* `+` is obsolete. just `= 1`